app/src/main/java/com/example/felipe/criminalintent/CrimeFragment.kt

```diff
@@ -9,28 +11,61 @@ import android.view.View
 import android.view.ViewGroup
 import android.widget.EditText
 import kotlinx.android.synthetic.main.fragment_crime.view.*
+import java.util.UUID
 
 class CrimeFragment : Fragment() {
+    companion object {
+        const val ARG_CRIME_ID = "crime_id"
+        val EXTRA_CRIME_ID = "${CrimeFragment::class.java.canonicalName}.crime_id"
+
+        fun newInstance(crimeId: UUID): CrimeFragment {
+            val args = Bundle()
+            args.putSerializable(ARG_CRIME_ID, crimeId)
+            val fragment = CrimeFragment()
+            fragment.arguments = args
+            return fragment
+        }
+    }
+
     private var crime = Crime()
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val crimeId = arguments?.getSerializable(ARG_CRIME_ID) as UUID
+        crime = CrimeController.getInstance().getCrime(crimeId) ?: crime
+    }
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        var view = inflater.inflate(R.layout.fragment_crime, container, false)
+        val view = inflater.inflate(R.layout.fragment_crime, container, false)
         initViews(view)
         return view
     }
 
     private fun initViews(view: View) {
         with(view) {
+            chkSolved.isChecked = crime.isSolved
+            chkPoliceRequired.isChecked = crime.isPoliceRequire
+            titleEditText.setText(crime.title)
+            btnCrime.text = crime.date.toString()
             chkSolved.setOnCheckedChangeListener { _, isChecked ->
                 crime.isSolved = isChecked
             }
-            btnCrime.text = crime.date.toString()
+            chkPoliceRequired.setOnCheckedChangeListener { _, isChecked ->
+                crime.isPoliceRequire = isChecked
+            }
             titleEditText.afterTextChangeListener {
                 crime.title = it.toString()
             }
         }
     }
 
+    fun onKeyDown() {
+        val intent = Intent()
+        intent.putExtra(EXTRA_CRIME_ID, crime.id.toString())
+        activity?.setResult(Activity.RESULT_OK, intent)
```

Comment:
maybe you could define an extension and call it as `activity?.finisActivity(Activity.RESULT_OK, intent)`